backend/app/models/types.py

```diff
@@ -59,4 +60,16 @@ def __eq__(self, other: Any) -> bool:
         return isinstance(other, EncryptedStr) and self.get_decrypted_value() == other.get_decrypted_value()
 
     def get_decrypted_value(self) -> str:
-        return security.decrypt_password(self)
+        decrypted_value: str = security.decrypt_password(self)
+        # if the decrypted_value matches the regex of a secrets provider,
+        # e.g. AWS Secrets Manager, GCP Secret Manager, Azure Key Vault
+        # get secret value from provider
+        try:
+            value: str = util.substitute_value_from_secret_store(decrypted_value)
+        except ImportError as e:
+            raise exceptions.SecretsModuleNotFoundError(e.msg)
```

Comment:
Nothing wrong with this but you should check the [tryceratops linter](https://github.com/guilatrova/tryceratops): it teaches good practices around exception handling. I learned a lot thanks to this tool!

Reply:
Looks very handy. I will give it a try, thanks!